opentelemetry-sdk/src/opentelemetry/sdk/configuration/__init__.py

```diff
@@ -161,3 +161,8 @@ def initialize_components():
     # some boilerplate in order to make sure current implementation does not
     # lock us out of supporting metrics later without major surgery.
     init_metrics(metric_exporters)
+
+
+class Configurator:
+    def configure(self):
+        initialize_components()
```

Comment:
This could be just a function as well right?

```suggestion
def configure():
    initialize_components()
```

and then in `sitecustomize.py` it would just be `entry_point.load()()`

Reply:
Hmno, the `load` method will yield the object that is pointed by the entry point value, in this case `opentelemetry.sdk.configuration:Configurator` which is a class. An entry point can point to a method, or some other object, but what this approach that we have been using is to declare an ABC in the API that mandates that certain method is implemented in order for the entry pointed classes to have it and run it when `sitecustomize` is invoked.

Reply:
I see, having the ABC would be helpful. Where would that definition go though? In `opentelemetry-instrumentation` or `opentelemetry-sdk`?

Does that mean that the other package would have to take the package that includes the ABC as a dependency?

(The `opentelemtery-sdk` to have its `Configurator` up to requirements and the `opentelmetry-instrumentation` to be able to validate classes it finds through the entry points)?

Reply:
No, wait, that ABC should be in `opentelemetry-instrumentation`. Sorry :sweat_smile: